images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

```diff
@@ -0,0 +1,371 @@
+        argument_spec = copy.deepcopy(AUTH_ARG_SPEC)
+        argument_spec.update(EVENT_ARG_SPEC)
+        return argument_spec
+
+    def execute_module(self):
+        """ Module execution """
+        self.client = self.get_api_client()#mostly auth
+
+        api_version = 'v1'
+        name_args = self.params.get('name')
+        namespace_args = self.params.get('namespace')
+        message = self.params.get('message')
+        reason = self.params.get('reason')
+        reportingComponent = self.params.get('reportingComponent')
+        event_type = self.params.get('type')
+        source = self.params.get('source')
+
+        definition = defaultdict(defaultdict)
+
+        def_meta = definition['metadata']
+        def_meta['name'] = self.params.get('name')
+        def_meta['namespace'] = self.params.get('namespace')
+
+        def_involvedObject = definition['involvedObject']
+        def_involvedObject['namespace'] = self.params.get('namespace')
+        def_involvedObject['apiVersion'] = self.params.get('apiVersion')
+        def_involvedObject['kind'] = self.params.get('kind')
+        def_involvedObject['name'] = self.params.get('name')
+        def_involvedObject['uid'] = self.params.get('uid')
+        def_involvedObject['resourceVersion'] = self.params.get('resourceVersion')
+        resource = self.find_resource('Event', 'v1', fail=True)#finds resource or gets it from the client api, looks for resrouce
+    # second call to find resource for involved object, set kind to arg of involved OBject kind
+    #rnewesource = self.find_resource('involvedObjectKind', 'v1', fail=True)
+
+    #this just establishes tha vars and seperates out the reason and count from returned cli object
+        priorCount = 1
+        try:
+            priorEvent=resource.get(name=def_meta['name'],
+                 namespace=def_meta['namespace'])
+            priorReason=priorEvent['reason']
+            print(priorReason)
+            print("current reason is %s" % reason)
+            priorCount = priorEvent['count']
+            print("the count from the prior event is %i" % priorCount)
+
+            if priorEvent is not None and priorReason != reason:
+                print("If reason changed %i" % priorCount)
+                now = datetime.datetime.now()
+                rfc = kubernetes.config.dateutil.format_rfc3339(now)
+                firstTimestamp=rfc
+                priorCount = 1
+                firstTimestamp=rfc
+                print("the value of the firstTimestamp new event is", firstTimestamp)
+            else:
+                #priorCount = 1
+                priorCount = priorCount + 1
+                print("the reason has changed", priorCount)
+        except openshift.dynamic.exceptions.NotFoundError:
+            pass
+
+        involvedObject_resourceVersion = "1"
+        involvedObject_uid = "1"
+        try:
+            totalEvent=resource.get(name=def_meta['name'], namespace=def_meta['namespace'])
```

Comment:
You're querying for the same event as the previous try/except block, and you're not changing any value of that event, so I think we can combine these two try/except blocks, e.g.:

```python
event = resource.get(name=def_meta['name'], namespace=def_meta['namespace'])

if event is not None:
    if event['reason'] != reason:
        # do all the event counting and timestamp tasks here
    
    # do all the involvedObj tasks here
```

Reply:
I considered this. Ultimately I really hate such nesting for readability, so I prefer to keep as is for clarity and separation.

Reply:
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

Reply:
See: Arrowhead Antipattern.

Reply:
There's no more or less nesting in the example provided, it's simply that you are currently spreading the same conditional logic in multiple places.